pr2eus/robot-interface.l

```diff
@@ -44,6 +44,58 @@
   (:time-to-finish ()
     (ros::ros-debug "[~A] time-to-finish ~A" ros::name-space time-to-finish)
     time-to-finish)
+  (:action-result-cb (msg)
+    "
+    display error messages from [depending on robots]/follow_joint_trajectory/result
+    "
+    ;; The following error messages will not appear for the first (send *ri* :angle-vector)
+    ;; To display it, use :wait-interpolation or (ros::spin-once groupname)
+    (let ((status-num (send (send msg :status) :status))
+          (status-word '(PENDING
+                         ACTIVE
+                         PREEMPTED
+                         SUCCEEDED
+                         ABORTED
+                         REJECTED
+                         PREEMPTING
+                         RECALLING
+                         RECALLED
+                         LOST))
+          (status-text (send (send msg :status) :text))
+          (error-code nil)
+          (error-word '(SUCCESSFUL
+                        INVALID_GOAL
+                        INVALID_JOINTS
+                        OLD_HEADER_TIMESTAMP
+                        PATH_TOLERANCE_VIOLATED
+                        GOAL_TOLERANCE_VIOLATED)))
+      ;; status
+      (cond ((not (or (= status-num 3) (= status-num 8)))
+             (ros::ros-error "joint trajectory status: ~A~%"
+                             (cons status-num (elt status-word status-num)))
+             (unless (string= "" status-text)
+               (ros::ros-error "~A~%" status-text)))
+            ((= status-num 8)
+             (ros::ros-warn "joint trajectory status: ~A~%"
+                            (cons status-num (elt status-word status-num)))
+             (unless (string= "" status-text)
+               (ros::ros-warn "~A~%" status-text))))
```

Comment:
You should not write the similar code twice.

How about this?
```
(let (log-func)
  (cond ((not (or (= status-num 3) (= status-num 8)))
               (setq log-func #'ros::ros-error))
            ((= status-num 8)
                (setq log-func #'ros::ros-warn)))
  (when log-func
    (funcall log-func "joint trajectory status: ~A~%"
                               (cons status-num (elt status-word status-num)))
                 (unless (string= "" status-text)
                   (funcall log-func "~A~%" status-text))))
```

Reply:
Thanks for your suggesting.
I'll use it.

> How about this?
```
(let (log-func)
  (cond ((not (or (= status-num 3) (= status-num 8)))
               (setq log-func #'ros::ros-error))
         ((= status-num 8)
           (setq log-func #'ros::ros-warn)))
  (when log-func
    (funcall log-func "joint trajectory status: ~A~%"
                               (cons status-num (elt status-word status-num)))
      (unless (string= "" status-text)
      (funcall log-func "~A~%" status-text))))
```